tests/test_unroll.py

```diff
@@ -0,0 +1,140 @@
+import inspect
+import ast
+import astor
+from ast_tools.transformers.loop_unroller import unroll_for_loops
+from ast_tools.passes import begin_rewrite, end_rewrite, loop_unroll
+
+
+def test_basic_unroll():
+    tree = ast.parse("""
+def foo():
+    for i in range(8, unroll=True):
```

Comment:
I think I would be more comfortable if we didn't hijack the name range.  This code isn't going to work without the rewrite so no need to pretend it's just normal python.

I would vastly prefer something like:
```
for i in ast_tools.macros.RANGE(8):
```
Where `ast_tools.macros.RANGE` is a real class that holds a `range` so that the code works with or without the rewrite.  

`visit_For` could be changed to:

```Python
def visit_For(self, node):
    try:
        range_object = ...
    except:
        is_constant = False
    else:
        is_constant = True
    if is_constant and instance(range_object, RANGE):
        # unroll
    else:       
        return self.generic_visit(node)
```

Reply:
`RANGE` could have a keyword argument `unroll` and could complain when initialized with `unroll=True` (as it should have been removed by a rewrite).  Or perhaps better we use `expand`  instead of `unroll` and have an `ast_tools.macros.MACRO` base class which has an `expand`  argument to init.